### ISSUE TITLE
Cannot link libraries for target and GMock

### ISSUE THREAD

**Issue by rob-meiner** [fada-catec/amcl3d#1]
Dear Colleagues,  

I'm having trouble building this in my catkin workspace.  I've successfully installed and built ```octomap_ros``` and ```octomap_msgs```, but now I'm confronting this error message:

```
  Could not find a package configuration file provided by "GMock" with any of
  the following names:

    GMockConfig.cmake
    gmock-config.cmake

  Add the installation prefix of "GMock" to CMAKE_PREFIX_PATH or set
  "GMock_DIR" to a directory containing one of the above files.  If "GMock"
  provides a separate development package or SDK, be sure it has been
  installed.


CMake Error at /home/user/catkin_ws/src/amcl3d/amcl3d/CMakeLists.txt:55 (target_link_libraries):
  Cannot specify link libraries for target
  "/home/user/catkin_ws/devel/lib/liboctomap.so" which is not built by this
  project.
```
I have noticed that ```GMock``` is now ```googletest``` [https://github.com/google/googletest](url), which appears to be split into ```googlemock``` and ```googletest```.  I don't know what the old ```GMock``` package looked like, so I'm not sure how to modify the ```amcl3d/CMakeLists.txt``` file appropriately.

I installed ```octomap``` with ```sudo apt-get install ros-melodic-octomap```, per the ros octomap page [https://wiki.ros.org/octomap](url) and modified the ```amcl3d/CMakeLists.txt``` and ```package.xml``` per the instructions on that ros page.

I would appreciate any help.

Thanks, Mark

**Comment by palomacf15** [comment]
Mr. Rob-meiner,

First of all, thank you for your interest in our package.

Regarding your installation questions, Gmoke jumps as a warning when it is not installed correctly. In the .traviz.yml file you can find a way to install it.
Regarding octomap, you must install the octomap-ros package. 
Be sure to do the correct installation and proper compilation as follows:

- sudo apt-get install ros-$ROS_DISTRO-octomap-ros
- mkdir -p ~/catkin_ws/src
- cd ~/catkin_ws/src
- catkin_init_workspace
- git clone https://github.com/fada-catec/amcl3d.git
- cd ~/catkin_ws
- catkin_make
- source devel/setup.bash

If you still have any problems with this, please pass on all the information that comes out of your terminal when you do the catkin_make. 

We tests this package in ubuntu 16.04 with ros kinetic.

**Comment by rob-meiner** [comment]
Dear pcarrascof,

Thanks for the reply and direction on how to fix the GMock problem.  I'll dig into that the next time I rebuild the workspace.

I did get this package to build, after I cloned and built all of the octomap packages, including octomap_ros, octomap_msgs, octomap (from source, not apt), and octomap_server.  After I had built them, I installed this package and was able to successfully build it with catkin tools.

I'm using Ubuntu 18.04 with ros Melodic.

Thanks again,
Mark
